taker-electron/src/main/endpoint.ts

```diff
@@ -0,0 +1,69 @@
+import { net } from "electron";
+import log from "electron-log";
+import nodenet from "node:net";
+
+const DEFAULT_PORT = 7113;
+
+export class Endpoint {
+    public url: string;
+
+    constructor(public port: number) {
+        this.url = `http://127.0.0.1:${port}`;
+    }
+
+    wait(onReady: (endpoint: string) => void) {
+        const alive = (timeout: number) => {
+            log.info(`Probing if ItchySats is alive at ${this.url}`);
+            const request = net.request(this.url);
+            request.on("response", () => onReady(this.url));
+            request.on("error", (error) => {
+                log.warn("Could not connect to ItchySats.");
+                log.error(error);
+                setTimeout(alive, timeout, timeout + timeout);
+            });
+            request.end();
+        };
+
+        setTimeout(alive, 500, 500, 500);
```

Comment:
it also would be good to extract the (arbitrary?) `500` to a const - is this in miliseconds?

Reply:
yes this is in miliseconds.